Lib/test/test_bytes.py

```diff
@@ -1889,6 +1889,41 @@ def __index__(self):
             with self.assertRaises(IndexError):
                 self._testlimitedcapi.sequence_setitem(b, 0, Boom())
 
+        class BoomContinued:
+            def __index__(self):
+                nonlocal new_ba
+                # Clear the original bytearray, mutating it during index assignment.
+                # If the internal buffers are held over this operation, they become dangling
+                # However, this will fail a bounds check as above (as the clear sets bounds to zero)
+                ba.clear()
+                # At this moment, the bytearray potentially has a dangling pointer
+                # Create a new bytearray to catch any writes
+                new_ba = bytearray(0x180)
+                # Ensure bounds check passes
+                ba.extend([0] * 0x180)
+                return 0
+
+        with self.subTest("skip_bounds_safety"):
+            new_ba: bytearray
```

Comment:
Use a global `new_ba` instead of a nonlocal one, or just assign it as an attribute of `BoomContinued` (or does it not work?)

Reply:
It should work. I chose to use a separate variable to make it more clear how the crash operates, and nonlocal to keep it specific to the test function instead of the global scope.